StreetHawk/Classes/Notification/Publish/SHApp+Notification.m

```diff
@@ -336,6 +339,24 @@ - (NSString *)apnsDeviceToken
     return [[NSUserDefaults standardUserDefaults] objectForKey:APNS_DEVICE_TOKEN];
 }
 
+- (void)setSHPushDenied
+{
+    if([self currentInstall].suid == nil){
+        return;
+    }
+
+    BOOL shouldSHPushBeDenied = [[UIApplication sharedApplication] currentUserNotificationSettings].types == UIUserNotificationTypeNone;
+    NSUserDefaults *preferences = [NSUserDefaults standardUserDefaults];
+    
+    if ([preferences objectForKey:SH_PUSH_DENIED_FLAG] == nil || [preferences boolForKey:SH_PUSH_DENIED_FLAG] != shouldSHPushBeDenied)
```

Comment:
One last thing is:
in your another pull request about location permission, you send tag when location module is not available in SHApp.m. 
Shall you do same thing in this push permission tag?

Reply:
OK, I see your SHApp.m change for the tag. 
There is a question:
Here you have a preferences saving previous sending tag, and only send sh_push_denied when permission change. However, in SHApp, that is called every often.